internal/database/actions.go

```diff
@@ -38,6 +42,60 @@ func ListMeasurements(sess sqlbuilder.Database, resultID int64) ([]MeasurementUR
 	return measurements, nil
 }
 
+type MeasurementWithInput struct {
+	Input string `json:"input"`
+}
+
+func GetMeasurementJSON(sess sqlbuilder.Database, measurementID int64) (map[string]interface{}, error) {
+	var (
+		measurement MeasurementURLNetwork
+		msmtJSON map[string]interface{}
+	)
+
+	req := sess.Select(
+		db.Raw("urls.*"),
+		db.Raw("measurements.*"),
+	).From("measurements").
+		LeftJoin("urls").On("urls.url_id = measurements.url_id").
+		Where("measurements.measurement_id= ?", measurementID)
+
+	if err := req.One(&measurement); err != nil {
+		log.Errorf("failed to run query %s: %v", req.String(), err)
+		return nil, err
+	}
+	reportFilePath := measurement.Measurement.ReportFilePath
+	if (measurement.URL.URL.Valid == false) {
```

Comment:
You should add a comment explaining why you're performing this check. (I assume this has to do with the fact that the measurement may or may not already be in explorer.)

Reply:
The `Valid` method is part of the standard `Nullable` `sql` objects. It's used to be able to make a query on a column where the value may be set to a particular value or to `NULL`.

This is checking if the measurement in question has a URL input. If there is no input then we don't need to seek to the right line offset because it's a single entry report file and we can just read it and return it.

Reply:
> This is checking if the measurement in question has a URL input. If there is no input then we don't need to seek to the right line offset because it's a single entry report file and we can just read it and return it.

Yes, this is the part that needs to be documented. It is a bit of information that otherwise I'd be missing.

Reply:
I added docs already: https://github.com/ooni/probe-cli/pull/53/commits/e0e12cdbca06aa5b5193467f51077af193d712c9#diff-33dda6bf2d32805bdafa67dc4c47250eR65

Reply:
👍